serverlessworkflow/modules/ROOT/pages/cloud/enhanced-service-discovery.adoc

```diff
@@ -0,0 +1,136 @@
+2022-08-23 12:21:06,043 INFO  [org.kie.kog.add.qua.k8s.SwKubeResource] (main) Connected to kubernetes cluster v1.23.4, current namespace is serverless-workflow-greeting-quarkus ---- Resource name is greeting-quarkus-cli
+2022-08-23 12:21:06,045 INFO  [org.kie.kog.add.qua.k8s.KnativeResourceDiscovery] (main) Trying to adapt kubernetes client to knative
+2022-08-23 12:21:06,316 INFO  [org.kie.kog.add.qua.k8s.KnativeResourceDiscovery] (main) Found Knative endpoint at http://greeting-quarkus-cli.serverless-workflow-greeting-quarkus.10.99.154.147.sslip.io
+2022-08-23 12:21:06,316 DEBUG [org.kie.kog.add.qua.k8s.SWDiscoveryConfigSourceInterceptor] (main) Discovered resource greeting-quarkus-cli - url http://greeting-quarkus-cli.serverless-workflow-greeting-quarkus.10.99.154.147.sslip.io
+2022-08-23 12:21:06,375 INFO  [org.kie.kog.sw.AppLifecycle] (main) The application is starting
+2022-08-23 12:21:06,382 INFO  [org.kie.kog.add.qua.mes.com.QuarkusKogitoExtensionInitializer] (main) Registered Kogito CloudEvent extension
+2022-08-23 12:21:06,460 INFO  [io.quarkus] (main) sw-service-discovery 1.0.0-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 2.360s. Listening on: http://0.0.0.0:8080
+2022-08-23 12:21:06,462 INFO  [io.quarkus] (main) Profile prod activated.
+2022-08-23 12:21:06,462 INFO  [io.quarkus] (main) Installed features: [cache, cdi, hibernate-validator, jackson-jq, kogito-addon-kubernetes-extension, kogito-addon-messaging-extension, kogito-processes, kogito-serverless-workflow, kubernetes, kubernetes-client, openshift-client, qute, reactive-routes, rest-client, rest-client-jackson, resteasy, resteasy-jackson, smallrye-context-propagation, smallrye-openapi, smallrye-reactive-messaging, smallrye-reactive-messaging-http, vertx]
+----
+
+[NOTE]
+====
+Note that the SWUrl was translated to `http://greeting-quarkus-cli.serverless-workflow-greeting-quarkus.10.99.154.147.sslip.io` +
+when the application started, and will be used at runtime any time it needs it.
+====
+
+== Discovery Precedence
+
+Depending on the resource to be discovered, there are some paths to follow, please see the following image for more details:
+
+image::cloud/sw-discovery-flow.jpg[]
+
+
+== Startup time: discovery enabled x disabled
+
+As the Service Discovery scans the Quarkus Configurations during the startup, it can cause a small delay as shown below:
+
+[tabs]
+====
+Service Discovery Enabled::
++
+[source,shell]
+----
+(main) sw-service-discovery 1.0.0-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 2.360s. Listening on: http://0.0.0.0:8080
+----
+Service Discovery Disabled::
++
+[source,shell]
+----
+(main) sw-service-discovery 1.0.0-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 1.507s. Listening on: http://0.0.0.0:8080
+----
+====
+
+[IMPORTANT]
+====
+When using this feature, you need to balance if the delayed startup time is something
+that your application can afford.
```

Comment:
Is it possible to implement lazy load? So the first time the application is called, the engine performs the discovery? If so we can have a new JIRA :P

Reply:
I am not sure, since the QuarkusConfig with interceptor is called at static init and when the system property is called for the first time.
I've opened this jira for further investigation.
https://issues.redhat.com/browse/KOGITO-7873